src/lib/stitching/exchange/stitching.ts

```diff
@@ -248,6 +250,82 @@ export const exchangeStitchingEnvironment = ({
             })
           },
         },
+        artworkOrEditionSet: {
+          fragment: gql`
+            ... on CommerceLineItem {
+              artworkId
+              editionSetId
+            }
+          `,
+          resolve: async (parent, _args, context, info) => {
+            const artworkId = parent.artworkId
+            const editionSetId = parent.editionSetId
```

Comment:
An order will always have an `artworkId`, but if it also has an `editionSetId`, then we know the order is actually on the corresponding edition set.